mobile/src/features/WalletManager/wallet-manager.ts

```diff
@@ -379,8 +379,21 @@ export class WalletManager {
       .then((tuples) => tuples.map(([_, walletMeta]) => walletMeta))
       .then((walletMetas) => walletMetas.filter(isWalletMeta)) // filter corrupted wallet metas
 
+    // Update walletMetas$ immediately with all metadata from storage
+    // This ensures hasWallets is accurate before wallets are fully loaded
+    // (which may involve slow network calls)
+    const allMetas = new Map(this.#walletMetas$.value)
```

Comment:
owww boy that's why sometimes it was showing the create wallet funnel (1st run) as if there was no wallet

nice catch! tricky one